packages/dynamite/dynamite/lib/src/openapi_builder.dart

```diff
@@ -38,6 +39,21 @@ class OpenAPIBuilder implements Builder {
 
   @override
   Future<void> build(final BuildStep buildStep) async {
+    try {
+      final result = await helperVersionCheck(buildStep);
+
+      if (result.messages.isNotEmpty) {
+        log.info(result.messages);
+      }
+
+      if (result.hasFatal) {
+        log.warning(result.messages);
+        throw Error();
+      }
+    } on Exception catch (exception) {
+      log.warning(exception);
+    }
```

Comment:
```suggestion
    final result = await helperVersionCheck(buildStep);

    if (result.messages.isNotEmpty) {
      if (result.hasFatal) {
        log.severe(result.messages);
        return;
      } else {
        log.info(result.messages);
      }
    }
```
Removes duplicate messages, `Instance of 'Error''` message and early exit if the constraints are wrong.